### PR TITLE
Add deleteAll to the orm

### PR DESCRIPTION
**Pull request by chkeita**
## Summary of the Pull Request

Introduce a DeleteAll operation to speedup the deletion of multiple entities.

The new method is able to delete 13404 Nodemessges in 4s

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/orm/Orm.cs [hunk 1]
```diff
@@ -134,6 +141,57 @@ public IAsyncEnumerable<T> SearchByRowKeys(IEnumerable<string> rowKeys)
...
+            var tableClient = await GetTableClient(typeof(T).Name);
+            var pages = tableClient.QueryAsync<TableEntity>(query, select: new[] { "PartitionKey, RowKey" });
+
+            var requests = await pages
+                .Chunk(MAX_TRANSACTION_SIZE)
+                .Where(chunk => chunk.Count > 0)
+                .Select(chunk =>
+                    chunk.Select(e => new TableTransactionAction(TableTransactionActionType.Delete, e))
+                )
+                .Select(t => tableClient.SubmitTransactionAsync(t))
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/orm/Orm.cs:179]: I would merge these two `Select`s:

```suggestion
                .Select(chunk => {
                    var t = chunk.Select(e => new TableTransactionAction(TableTransactionActionType.Delete, e));
                    return tableClient.SubmitTransactionAsync(t));
                });
```

File: src/ApiService/ApiService/onefuzzlib/orm/Orm.cs [hunk 2]
```diff
@@ -134,6 +141,57 @@ public IAsyncEnumerable<T> SearchByRowKeys(IEnumerable<string> rowKeys)
...
+                .ToArrayAsync();
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/orm/Orm.cs:180]: small, but using `ToList` does less copying if we don't need an array
```suggestion
                .ToListAsync();
```

File: src/ApiService/ApiService/onefuzzlib/Utils.cs [hunk 3]
```diff
@@ -15,3 +15,22 @@ public static class ObjectExtention {
...
+    public static async IAsyncEnumerable<List<TSource>> Chunk<TSource>(this IAsyncEnumerable<TSource> source, int size) {
+
+        var enumerator = source.GetAsyncEnumerator();
+        List<TSource> result = new List<TSource>(size);
+        while (await enumerator.MoveNextAsync()) {
+            if (result.Count == size) {
+                yield return result;
+                result = new List<TSource>(size);
+            }
+            result.Add(enumerator.Current);
```
Comment (Porges) [inline src/ApiService/ApiService/onefuzzlib/Utils.cs:29]: Do this in the other order so it `yield`s the result as soon as it is ready?

```suggestion
            result.Add(enumerator.Current);
            
            if (result.Count == size) {
                yield return result;
                result = new List<TSource>(size);
            }
```

**Review by Porges — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/Utils.cs [hunk 4]
```diff
@@ -15,3 +15,22 @@ public static class ObjectExtention {
     public static Async.Task IgnoreResult<T>(this Async.Task<T> task)
         => task;
 }
+
+public static class IAsyncEnumerableExtension {
+    public static async IAsyncEnumerable<List<TSource>> Chunk<TSource>(this IAsyncEnumerable<TSource> source, int size) {
+
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/Utils.cs:21]: We should throw if `size < 1`

File: src/ApiService/ApiService/onefuzzlib/orm/Orm.cs [hunk 5]
```diff
@@ -134,6 +141,57 @@ public IAsyncEnumerable<T> SearchByRowKeys(IEnumerable<string> rowKeys)
...
+                    }
+                )
+            );
+
+            var tableClient = await GetTableClient(typeof(T).Name);
+            var pages = tableClient.QueryAsync<TableEntity>(query, select: new[] { "PartitionKey, RowKey" });
+
+            var requests = await pages
+                .Chunk(MAX_TRANSACTION_SIZE)
+                .Where(chunk => chunk.Count > 0)
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/orm/Orm.cs:175]: It shouldn't be possible for `chunk.Count == 0`

**Review by tevoinea — APPROVED** [review]

**Pull request merged**